src/crypto/tls/handshake_client.go

```diff
@@ -146,30 +146,60 @@ func (c *Conn) makeClientHello(minVersion uint16) (*clientHelloMsg, clientKeySha
 			hello.cipherSuites = append(hello.cipherSuites, defaultCipherSuitesTLS13NoAES...)
 		}
 
-		curveID := config.curvePreferences()[0]
-		if scheme := curveIdToCirclScheme(curveID); scheme != nil {
-			pk, sk, err := generateKemKeyPair(scheme, curveID, config.rand())
-			if err != nil {
-				return nil, nil, fmt.Errorf("generateKemKeyPair %s: %w",
-					scheme.Name(), err)
+		hello.keyShares = []keyShare{}
```

Comment:
We know the length of keyShares (after moving it to line 155).
```suggestion
		hello.keyShares = make([]keyShare, 0, len(curveIDs))
```